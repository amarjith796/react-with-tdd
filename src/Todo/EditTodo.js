import React, { Component } from 'react';
import Formsy from 'formsy-react';
import PropTypes from 'prop-types';
import MyInput from './MyInput';
import { addValidationRule } from 'formsy-react';
function isNotBlankString(values, value) {
    return value && value.trim().length > 0;
}

addValidationRule('isNotBlankString', isNotBlankString);
class EditTodo extends Component {
    constructor(props) {
        super(props);
        this.state = {
            textInput: this.props.editText,
            canSubmit: false
        }
        this.disableButton = this.disableButton.bind(this);
        this.enableButton = this.enableButton.bind(this);
    }
    componentWillReceiveProps(nextProps, prevState) {
        if (this.state.textInput != nextProps.editText) {
            this.setState({
                textInput: nextProps.editText
            })
        }
    }
    handleOnchange = (event) => {
        this.setState({
            textInput: event.target.value
        });
    }
    disableButton() {
        this.setState({ canSubmit: false });
    }

    enableButton() {
        this.setState({ canSubmit: true });
    }

    submit = (model) => {
        const { editText, handleUpdateTodo } = this.props;
        handleUpdateTodo(editText, model.text)
    }
    render() {
        const { editText, handleUpdateTodo } = this.props;
        const { textInput } = this.state;
        return (
            <div style={{ display: 'flex', width: '60vw' }}>
                {/* <input type='text' style={{ flex: 1 }} value={textInput} onChange={this.handleOnchange} /> */}
                <Formsy style={{ flex: 1, display: 'flex' }} onValidSubmit={this.submit} onValid={this.enableButton} onInvalid={this.disableButton}>
                    <MyInput
                        name="text"
                        validations="isNotBlankString"
                        value={textInput}
                        validationError="Please enter the taskname"
                        required
                    />
                    <button type="submit" disabled={!this.state.canSubmit} style={{ padding: '10px', background: 'steelblue', border: '1px solid steelblue', cursor: 'pointer' }} >Update</button>
                </Formsy>
            </div>);
    }
}
EditTodo.propTypes = {
    editText: PropTypes.any,
    handleUpdateTodo: PropTypes.func
};

export default EditTodo;
