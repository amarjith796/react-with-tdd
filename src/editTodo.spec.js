import React from 'react';
import { shallow } from 'enzyme';
import EditTodo from './Todo/EditTodo';
describe('EditTodo', () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<EditTodo editText={""} handleUpdateTodo={jest.fn()} />));
    it('should render correctly', () => expect(wrapper).toMatchSnapshot());
    it('should render a <div />', () => {
        expect(wrapper.find('div').length).toBeGreaterThanOrEqual(1);
    });
});