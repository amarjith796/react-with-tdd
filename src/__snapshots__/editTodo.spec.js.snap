// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTodo should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditTodo
    editText=""
    handleUpdateTodo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          type="text"
          value=""
        />,
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "steelblue",
              "border": "1px solid steelblue",
              "cursor": "pointer",
              "padding": "10px",
            }
          }
        >
          Update
        </button>,
      ],
      "style": Object {
        "display": "flex",
        "width": "500px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "style": Object {
            "flex": 1,
          },
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Update",
          "onClick": [Function],
          "style": Object {
            "background": "steelblue",
            "border": "1px solid steelblue",
            "cursor": "pointer",
            "padding": "10px",
          },
        },
        "ref": null,
        "rendered": "Update",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
            type="text"
            value=""
          />,
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "steelblue",
                "border": "1px solid steelblue",
                "cursor": "pointer",
                "padding": "10px",
              }
            }
          >
            Update
          </button>,
        ],
        "style": Object {
          "display": "flex",
          "width": "500px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "style": Object {
              "flex": 1,
            },
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Update",
            "onClick": [Function],
            "style": Object {
              "background": "steelblue",
              "border": "1px solid steelblue",
              "cursor": "pointer",
              "padding": "10px",
            },
          },
          "ref": null,
          "rendered": "Update",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
